File: MyWorkItemFrontend/src/types/apiResponse.ts
export enum ResponseCode {
  Success          = 0,
  SystemError      = 1,
  NotFoundId       = 2,
  ValidationError  = 3,
  Unauthorized     = 4,
  Forbidden        = 5,
  DuplicateEntry   = 6,
  InvalidOperation = 7,
}

export interface ApiResponseBase {
  code: ResponseCode;
  message: string;
}

export interface ApiResponse<T> extends ApiResponseBase {
  content: T | null;
}

File: MyWorkItemAPI/MyWorkItemAPI/Common/ApiResponse.cs
namespace MyWorkItemAPI.Common;

public class ApiResponse<T> : ApiResponseBase
{
    public T? Content { get; set; }

    public static ApiResponse<T> Success(T content) => new()
    {
        Code = ResponseCode.Success,
        Message = "Success",
        Content = content
    };

    public static ApiResponse<T> Fail(ResponseCode code, string message) => new()
    {
        Code = code,
        Message = message
    };
}
